prompting-client/src/snapd_client/interfaces/mod.rs

```diff
@@ -32,43 +29,9 @@ pub trait SnapInterface: fmt::Debug + Clone {
 
     fn prompt_to_reply(prompt: Prompt<Self>, action: Action) -> PromptReply<Self>;
 
-    fn map_ui_input(&self, prompt: Prompt<Self>, meta: Option<SnapMeta>) -> UiInput<Self>;
+    fn map_ui_input(&self, prompt: Prompt<Self>, meta: Option<SnapMeta>) -> Result<UiInput<Self>>;
 
     fn map_ui_reply(&self, reply: Self::UiReply) -> PromptReply<Self>;
-
-    async fn try_get_reply_from_ui(
```

Comment:
this should have been removed when we moved to GRPC but it was missed. As it is a trait method on a public trait it didn't get picked up by dead code analysis. I'll make the trait `pub(crate)` actually in order to catch this sort of thing in future.

Reply:
ah, looks like we can't actually make it `pub(crate)` as we need to make use of it in the integration tests. I've removed another unused method though.